[syCHEN1645] iP

syCHEN1645 opened the pull request.

syCHEN1645 committed:
Finish level-0

syCHEN1645 committed:
Finish level-1

Implement new tone for Rook
Implement method ChatEcho to echo a line
Implement method ChatGreeting to greet

syCHEN1645 committed:
Finish level-2

Add a static list of activities
Add method ChatAddList() to add and list activities

syCHEN1645 committed:
Add new class Activity

syCHEN1645 committed:
Finish level-3

Amend some naming errors
Improve code structure by moving while loop from methods to main
Add new methods to check if a message is a proper command
Add new methods for Rook to mark & unmark activities

syCHEN1645 committed:
Finish level-3

syCHEN1645 committed:
Merge branch 'master' of https://github.com/syCHEN1645/ip

syCHEN1645 committed:
Restructure OO for Level-4

Rename Activity class to Todo class
Add Task class, Event class, Deadline class
Todo extends Task, (Deadline, Event) extends Todo
Delete outdated code in comment

syCHEN1645 committed:
Add Command enum

Compile all command keywords into Command enum
Break down methods in Rook to meet coding standard
Make minor changes

syCHEN1645 committed:
Finish Level-4

Add and modify methods to add different types of tasks
Format Rook's output
Fix output bugs

syCHEN1645 committed:
Update text-ui-test

Add simple test cases for "list" and "bye"

SahejAgarwal05 commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,24 @@
+public enum Command {
```
Smart move I wish I thought of that

SahejAgarwal05 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,37 @@
+    public String getEndTime() {
+        return endTime;
+    }
+
```
should add command documentation

SahejAgarwal05 commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+    static final String LOGO = """
+                \t___ ___ ___
+                \t| |_| |_| |
+                \t \\       /
```
Very Readable code

SahejAgarwal05 commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+        String description = parts[0].strip();
+        String byTime = "";
+        if (parts.length > 1) {
+            byTime = parts[1].strip();
```
Good Job following everything to the letter

SahejAgarwal05 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+    public Task(String description) {
+        setDescription(description);
+        setDone(false);
+    }
```
Good function names

SahejAgarwal05 reviewed (commented):
GOod Job

Cheng-Zhiyuan commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+        System.out.println(PARTITION);
+    }
+
+    private static void chatGreeting() {
```
Great modularizing of methods

Cheng-Zhiyuan commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+        System.out.println(PARTITION);
+    }
+
+    private static boolean isValidEventCommand(String message) {
```
Great boolean naming convention

Cheng-Zhiyuan commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+            result += digits[i];
+        }
+        return result;
+    }
```
Perhaps can use constants instead of numbers

Cheng-Zhiyuan commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,272 @@
+        tasks.get(index - 1).setDone(true);
+        System.out.println(PARTITION);
+        System.out.println(tasks.get(index - 1));
+        System.out.println("My lord, Activity " + index + " has been marked under your command!");
```
Maybe we can make this into a method?

Cheng-Zhiyuan reviewed (commented):
Overall it the code is very good and clear!

syCHEN1645 committed:
Add class RookException and child classes

Add child classes InvalidInfoException, MissingInfoException and MissingKeywordException
Implemented throwing exceptions for method chatMarkDone

syCHEN1645 committed:
Finish Level-5

Throw exception in methods of chat commands
Add error messages for child classes of RookException
Removed some old unused methods

syCHEN1645 committed:
Level-5

Merge branch-Level-5 to master branch

syCHEN1645 committed:
Organise into Packages

Put files into packages

syCHEN1645 committed:
Add Delete Function

Add method chatDelete
Add new command "delete"

syCHEN1645 committed:
Update test file

Update test file to test more functions

syCHEN1645 committed:
Add Class FileManager

Add new class FileManager to write and read from a saved file.
Adjust class Task for saving information into file.

syCHEN1645 committed:
Create Save File

Update constructor to create a save file upon called.
Modified child classes of Task accordingly.

syCHEN1645 committed:
Load Saved Data

Implement load data methods in class FileManager.
Integrate FileManager into Rook's main.
Cannot update isDone status yet.

syCHEN1645 committed:
Merge Branch 'branch-Level-6'

syCHEN1645 committed:
Merge Branch 'branch-Level-7'

syCHEN1645 committed:
Merge Branch 'master' into branch-Level-6

syCHEN1645 committed:
Merge Branch 'master' into branch-Level-7

syCHEN1645 committed:
Debug for Level-6 and Level-7

Implement delete methods in save file.
Repair output bugs.

syCHEN1645 committed:
Update gitignore

Ignore save file.

aditichadha1310 commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,376 @@
+
+    private static void loadSavedData() {
+        tasks = fileManager.readTask();
+//        if (tasks == null) {
+//            tasks = new ArrayList<>();
+//        }
```
You might want to remove commented out code.

aditichadha1310 commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,376 @@
+            throw new MissingInfoException();
+        }
+        // Case no "/by"
+        if (indexByTime == -1) {
```
Avoid usage of magic numbers

aditichadha1310 commented on src/main/java/Rook.java:
```diff
@@ -0,0 +1,376 @@
+        );
+    }
+
+    private static void chatAddEvent(String message) throws RookException {
```
avoid lengthy methods. try to keep each method to about 30 lines of code.

aditichadha1310 reviewed (commented):
overall, good work!

syCHEN1645 committed:
Remerge branch 'branch-Level-6'

In previous merge 'branch-Level-6' has no upstream.

syCHEN1645 committed:
Remerge branch 'branch-Level-7'

In previous merge 'branch-Level-7' has no upstream.

syCHEN1645 committed:
Improve Coding Standard

Correct package names to using lower cases.
Reduce magic numbers.
Reduce lengths for methods in Rook.

syCHEN1645 committed:
Implement Method printLines Using Varargs

syCHEN1645 committed:
Fix Bugs in Mark and Unmark Methods

syCHEN1645 committed:
Create New Class CommandManager

CommandManager contains methods to extract commands.
Fix minor bugs.

syCHEN1645 committed:
Create Class CommandManager

CommandManager contains methods to translate commands.

syCHEN1645 committed:
Implement Find

Add command 'find' and implement related methods.

syCHEN1645 committed:
Revert "Implement Find"

This reverts commit 05bb3f922ac048b75e21f3bec7ac3d0bf5a17695.

syCHEN1645 committed:
Implement Find

Create new command 'find' to find tasks that match in description.
Move Class Command from command package to ui package.

syCHEN1645 committed:
Merge branch 'branch-Level-9'

syCHEN1645 committed:
Implement LocalDateTime

Both LocalDateTime and String can be used for time in Class Event.

syCHEN1645 committed:
Implement More LocalDateTime

LocalDateTime can be used in Deadline, too.
String can still be used as time, but a reminder will be given.

syCHEN1645 committed:
Merge branch 'branch-Level-8'

Implement LocalDateTime in branch-Level-8.

syCHEN1645 committed:
Add JavaDoc Comments

Comment on Rook, Command, CommandManager, Task and its child classes.

syCHEN1645 committed:
More JavaDoc Comments

Comment on RookException and its child classes, FileManager.
Remove some codes that are commented out.

syCHEN1645 committed:
Merge branch 'branch-A-JavaDoc'

Comment on most public classes and methods, and some private and protected methods usnig JavaDoc.

syCHEN1645 committed:
Update README.md

Update user guide for Rook.
Introduce all user commands.

syCHEN1645 committed:
Improve README

syCHEN1645 committed:
Fix Output Bug

Fix bug which causes unnecessary printing of reminder messages in constructors of Deadline and Event.

syCHEN1645 committed:
Merge pull request #5 from syCHEN1645/branch-Practice-Merge-PR

Branch-Practice-Merge-PR is good to merge.

syCHEN1645 committed:
Update README.md
